roles/ceph-validate/tasks/check_iscsi.yml

```diff
@@ -0,0 +1,9 @@
+---
+- name: fail if bad chap settings
+  fail:
+    msg: "Mixing clients with CHAP enabled and disabled is not supported."
+  with_items: "{{ client_connections }}"
+  when:
+    - item.status == "present"
+    - item.chap != ''
+    - " '' in client_connections | selectattr('status', 'match', '^present$') | map(attribute='chap') | list"
```

Comment:
I think you don't need to write`'^present$'`.
`'present'` is enough since you are using `'match'` rather than `'search'`